### ISSUE TITLE
Expectations are not evaluated on mock objects passed via @depends

### ISSUE THREAD

**Issue by sebastianbergmann** [sebastianbergmann/phpunit#1821]
## Example

```php
<?php
class Foo
{
    public function bar()
    {
    }
}

class Test extends PHPUnit_Framework_TestCase
{
    public function testOne()
    {
        $mock = $this->getMock(Foo::class);
        $mock->expects($this->once())->method('bar');

        $mock->bar();

        return $mock;
    }

    /**
     * @depends testOne
     */
    public function testTwo(Foo $mock)
    {
        $mock->bar();
    }
}
```

## Actual output

```
PHPUnit 4.8.3 by Sebastian Bergmann and contributors.

..

Time: 64 ms, Memory: 4.75Mb

OK (2 tests, 1 assertion)
```

## Expected output

```
PHPUnit 4.8.3 by Sebastian Bergmann and contributors.

.F

Time: 64 ms, Memory: 4.75Mb

There was 1 failure:

1) Test::testTwo
Foo::bar() was not expected to be called more than once.

/home/sb/phpunit-mock-depends/Test.php:26

FAILURES!
Tests: 1, Assertions: 1, Failures: 1.
```

**Comment by nicolas-grekas** [comment]
Couldn't this be considered as a bug fix on 4.8? It's a shame we have to use php 5.6 to get these assertions correctly verified...

**Comment by nicolas-grekas** [comment]
Btw, this issue is also about mock objects passed via data collectors.
